pure-pulsars/src/cmds/rabbit_hole.py

```diff
@@ -0,0 +1,156 @@
+- Categories: Provide adjacent categories for further exploration. This should include 3 categories and their respective Wikipedia URLs that are relevant to the topic (but not about the topic itself).
+
+Output the summary in the following JSON format:
+{
+    "Intro": "Brief summary of the article's intro.",
+    "Sections": {
+        "Section Name": ["Sentence 1", "Sentence 2"],
+        "Section Name": ["Sentence 1", "Sentence 2"],
+    },
+    "Categories": [
+        {"Name": "Category 1", "URL": "Wikipedia URL"},
+        {"Name": "Category 2", "URL": "Wikipedia URL"},
+        {"Name": "Category 3", "URL": "Wikipedia URL"},
+    ],
+}
+
+Do not format the message with ```json```, as it is not neccessary. Return the JSON object as a string.
+"""
+
+genai.configure(api_key=os.environ["GEMINI_API_KEY"])
+model = genai.GenerativeModel("gemini-1.5-flash", system_instruction=sys_ins)
+
+
+def make_embed(summary: dict) -> Embed:
+    """Create an embed message for Wikipedia.
+
+    Args:
+    ----
+    summary (dict): The summary of the Wikipedia article.
+
+    Returns:
+    -------
+    discord.Embed: The embed message.
+
+    """
+    embed_msg = Embed(title=summary["Title"], description=summary["Intro"], url=summary["URL"], color=0xFFFFFF)
+    for section, sentences in summary["Sections"].items():
+        embed_msg.add_field(name=section, value="\n".join([f"- {sentence}" for sentence in sentences]), inline=False)
+    embed_msg.set_image(url=summary["Image"])
+    embed_msg.set_footer(text="To explore more Wikipedia topics, click one of the buttons below.")
+    return embed_msg
+
+
+class WikiButtons(discord.ui.View):
+    """Buttons for exploring more Wikipedia articles."""
+
+    def __init__(self, pages: list[Page]) -> None:
+        """Initialize the WikiButtons class.
+
+        Args:
+        ----
+        pages (list[Page]): The Wikipedia pages to explore.
+
+        """
+        super().__init__(timeout=None)
+        self.pages = pages
+        self.create_buttons()
+
+    def create_buttons(self) -> None:
+        """Create buttons for each Wikipedia article."""
+
+        def create_callback(page: Page) -> discord.ui.Button.callback:
+            async def button_callback(interaction: discord.Interaction) -> None:
+                await interaction.response.defer(thinking=True, ephemeral=True)
+                await rabbit_hole_helper(interaction, page)
+
+            return button_callback
+
+        for page in self.pages:
+            button = discord.ui.Button(label=page.title(), style=discord.ButtonStyle.green)
+            button.callback = create_callback(page)
+            self.add_item(button)
+
+
+async def rabbit_hole_helper(interaction: discord.Interaction, article: Page) -> None:
+    """Functions to help the rabbit hole."""
+    try:
+        # Generate a summary and return an embed message
+        text = article.extract(intro=False)
+        response = model.generate_content(text)
```

Comment:
This is an interesting way of getting the response from the AI, good catch on the try/except. *However*, I think there should be a note about the potential miss-information from using a LLM.